src/Markup/Avalonia.Markup.Xaml.Loader/AvaloniaXamlIlRuntimeCompiler.cs

```diff
@@ -178,13 +180,14 @@ static object LoadSreCore(string xaml, Assembly localAssembly, object rootInstan
             var clrPropertyBuilder = tb.DefineNestedType("ClrProperties_" + Guid.NewGuid().ToString("N"));
             var indexerClosureType = _sreBuilder.DefineType("IndexerClosure_" + Guid.NewGuid().ToString("N"));
 
+            bool compileBindingsByDefault = AppContext.TryGetSwitch(UseCompileBindingsByDefaultConfigSwitch, out var compileBindingsSwitchValue) && compileBindingsSwitchValue;
```

Comment:
Assembly containing the XAML file and the executable assembly might come from different projects with different settings. 
I think we should pass that switch using some `AvaloniaResource` or assembly metadata rather than `runtimeconfig.json` file.